app/javascript/extensible-components/lib.ts

```diff
@@ -0,0 +1,26 @@
+export interface IExtensionComponent {
+  action: string;
+  payload: any;
+}
+
+export interface IMiQApiCallback {
+  [propName: string]: Function;
+}
+
+export interface IExtensibleComponent {
+  extensibleComponent: any;
+  apiCallbacks: () => IMiQApiCallback;
+  renderCallbacks: () => IMiQApiCallback;
+}
+
+export const getItems = ManageIQ.extensions.getItems;
```

Comment:
I guess we'd also want to `export function subscribe` here ?

Generally, all the stuff we put into `ManageIQ.extensions` should be exported here, I think.